common-service-core/src/main/java/org/zowe/apiml/util/MapUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.util;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+public class MapUtils {
```

Comment:
What about write it like class library (not as "helper")?

public *final* class MapUtils {

*private MapUtils() {}*

public *static final* Map<String, String> flattenMap....

}

Reply:
I intended to preserve the class so it can be mocked and it's methods can be mocked or spied on to help with testing.

Reply:
check out the latest impl, I've modified the code to do this